File: mac/Snag/Workers/SnagController/Models/SnagLog.swift
import Foundation

struct SnagLog: Codable, Equatable, Hashable, Identifiable {
    var id: UUID = UUID()
    let timestamp: Date
    let level: String
    let message: String
    let tag: String?
    let details: [String: String]?
    
    enum CodingKeys: String, CodingKey {
        case timestamp, level, message, tag, details
    }
    
    init(timestamp: Date = Date(), 
         level: String, 
         message: String, 
         tag: String? = nil, 
         details: [String: String]? = nil) {
        self.timestamp = timestamp
        self.level = level
        self.message = message
        self.tag = tag
        self.details = details
        
    }
}

File: android/snag/src/main/java/com/snag/models/SnagLog.kt
package com.snag.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.util.Date

@Serializable
data class SnagLog(
    @SerialName("timestamp")
    val timestamp: Long = System.currentTimeMillis() / 1000,
    @SerialName("level")
    val level: String,
    @SerialName("message")
    val message: String,
    @SerialName("tag")
    val tag: String? = null,
    @SerialName("details")
    val details: Map<String, String>? = null
)
